src/java.desktop/share/classes/com/sun/beans/finder/MethodFinder.java

```diff
@@ -79,7 +78,7 @@ public static Method findMethod(Class<?> type, String name, Class<?>...args) thr
 
         try {
             Method method = CACHE.get(signature);
-            return (method == null) || isPackageAccessible(method.getDeclaringClass()) ? method : CACHE.create(signature);
+            return (method == null) ? method : CACHE.create(signature);
```

Comment:
This code looks wrong, meaning it was wrong already.
It is out of scope here to evaluate and fix that, so I submitted https://bugs.openjdk.org/browse/JDK-8344892